### ISSUE TITLE
Index for BHĀGAVATAPURĀṆA, Bombay edition.

### ISSUE THREAD

**Issue by funderburkjim** [sanskrit-lexicon/PWG#82]
Preparation of an index for this, as needed for #81 bombay edition.

@AnnaRybakovaT  
There are two volumes.
Here is a link to pdf for volume 1 (~600MB):
 https://drive.google.com/file/d/1FRW3NVbZZR04xoCF62MK2OwaTgVUfFBG/view?usp=drive_link

@Andhrabharati   This differs from the original link you provided, in that the pages have been rotated 90 degrees.

--------------------------------------------------

Anna, Andhrabharati prepared an [index for the Burnouf edition of BhP](https://github.com/sanskrit-lexicon/PWG/blob/master/pwgissues/issue81/Burnouf.BhP.index.txt)

Your task will be to construct a similar index for bombay edition.

The index will be used by an application which shows the page for a given skandha, adhyAya and verse.  For example, with the burnouf version : https://sanskrit-lexicon-scans.github.io/bhagp_bur/app1/?2,4,6

For now, just familarize yourself with the above.

**Comment by funderburkjim** [comment]
@Andhrabharati I don't yet understand this Indian version.  Perhaps you could provide an introduction that will help Anna get started.

**Comment by funderburkjim** [comment]
Here is the 'manifest'.  
```
Title: Bhagavatá-Purana: in 12 Books. (Printed with moveable type). 
By: with a Commentary by Sridharaswamin, called Srîtchâgavata bhâvârtha dêpikâ
Creation: Bombay 1860
Language: Sanskrit
Identifier: BSB-ID 991133463199707356
Identifier: BV021135362
Identifier: OCLC 162838650
URN: urn:nbn:de:bvb:12-bsb10211024-1
Media type: Book
Related Links
Related Details
(text/html)
OPAC
(text/html)
See also
MARCXML
(application/marcxml+xml)
RDF
(application/rdf+xml)
IIIF manifest
https://api.digitale-sammlungen.de/iiif/presentation/v2/bsb10211024/manifest
```

conjecture: the smaller Devanagari is Commentary by Sridharaswamin,.
And the larger Devanagari is the text of BhP.  Our index will be based on the larger Devanagari.

**Comment by Andhrabharati** [comment]
> @Andhrabharati I don't yet understand this Indian version.  Perhaps you could provide an introduction that will help Anna get started.
> 

Jim/Anna,

I would suggest starting with the skandhas 10-12, that were not in the Burnouf edition. (So, you need to share that as well, as modified by you.)

If this is agreed upon, I can put more of my thoughts about the task.

**Comment by funderburkjim** [comment]
A copy of volume 2 is: 
https://drive.google.com/file/d/1vdeZAfE8FFDSXE3EOrSMk1yLTHSHKtzg/view?usp=drive_link

**Comment by AnnaRybakovaT** [comment]
> I can put more of my thoughts about the task
Dear all,

I am waiting for an access to download the files with 2 volumes and your further recommendations regarding this task.

**Comment by AnnaRybakovaT** [comment]
> A copy of volume 2 is:

Dear Jim,
Thanks a lot, I have downloaded both volumes.

My first qustion is about numeration of pages. For example: the 1st volume,  book's page 1 is located on two pdf pages - 11 and 12. Does it have sence for our Index (I mean to use 1a and 1b, for example) - or just to put number 1?

**Comment by funderburkjim** [comment]
Anna, for our current purposes, the pdf page number is what is needed.  

You pointed out that the 'internal' pages appear to be in two parts. Although I don't understand the purpose of this,
It seems that the 'internal' page numbers 'start over' after the 'decorative-pages' at skandha breaks.

The Burnouf index has 6 fields; you could add a 7th field  'ipage' using '1a, 1b, etc' for these internal page numbers.
```
volume	page	sk.	adhy.	from v.	to v.
volume	page	ipage	sk.	adhy.	from v.	to v.
```

@Andhrabharati  Would this internal page number have some future use?

--------------------------------------------------------------

It might be useful to make a separate 'skandha_pages.txt' file such as
```
1 8 अथश्रीमद्भागवतेप्रथमस्कंध्ःप्रारभ्यते 
1 115 अथश्रीमद्भागवतेप्रथमस्कंध्ःसमाप्तः
```
There are some variations to this form, whose significance is unclear to me. For instance there are two parts to 10th skandha पूर्व उत्तर, and there is a final माहात्म्य section in volume 2.

------------------------------------------------------------------

There are some 'skew' pages.  Suggest you make a separate 'skew_pages.txt' file containing volume/page for these.  There may be some way to 'straighten' these.

-----------------------------------------------------------------

In the Burnouf edition, some verses appear partly at the bottom of one page and partly at the top of the next page.  @Andhrabharati used the pada designations  a,b,c,d  in which such a verse appear on two lines of the index.  I haven't noticed phenomenon in the Bombay edition, but it may happen.

**Comment by AnnaRybakovaT** [comment]
> The Burnouf index has 6 fields; you could add a 7th field 'ipage' using '1a, 1b, etc' for these internal page numbers.

Dear all,
I started with skandha 1 just to get used to the book and its structure. In a while I will focus on skandhas 10-12.
Could you kindly recommend me the easiest way - how to create a txt file with 7 fields or add one field to the Burnouf.BhP.index.txt file which consists of 6 fields. I tried to solve this topic by myself but I was confused by different options like tabs, semicolons, pandas.

**Comment by gasyoun** [comment]
>  how to create a txt file with 7 fields

tabs or semicolons, any permament way would sufficient

**Comment by funderburkjim** [comment]
@AnnaRybakovaT  You may have seen the term 'csv file'  (comma-separated-values).  Such a file is a useful way to represent  data which has a tabular form.  For example, suppose you are planning a vegetable garden, and are deciding how many of various plants you need to buy.
Say you want 5 tomato plants, 3 cucumber plants., and 6 green bean plants.  You could make a text file (using any text editor) like:
```
plant,count
tomato,5
cucumber,3
green bean,6
```
Such a file would be called a 'csv' file.  Here the comma character ',' separates the values on each line.  The first (optional) line contains (by convention) titles for the fields.

Or, you could create a 'colon-separated-values'  file.
```
plant:count
tomato:5
cucumber:3
green bean:6
```
For the data of this toy example, the 'value-separator' can be any character which does NOT occur within any field.  For instance the space character would NOT be a good value separator since the 'green bean' value  already has a space.

Some people are comfortable using spreadsheets  (such as Excel, or Google Sheets') to create tabular data files.  You can always save such files as 'tsv; (tab-separated-values) files or 'csv' files.  

To summarize, in our case pick one:
* create a csv file with a text editor, and use either (a) comma or (b) colon as the separator
   * You can also, for readability, add extra spaces around the separator character:
     `tomato    ,   5`  
* create a spreadsheet file with excel or google sheets.  When you are done, save the result as a tsv file.

Unless you are familiar with using spread sheets, I suggest you make a colon-separated 
file with a text editor.  Also include the first 'title' line

When you create a file with the first 5 lines,   upload it so we can check the form.

A search for 'CSV file' will bring up various introductions to this topic. CSV files are ubiquitous, so it is worthwhile to understand the idea of a csv file (and it's near cousins such as tsv files).

**Comment by AnnaRybakovaT** [comment]
> When you create a file with the first 5 lines, upload it so we can check the form.

Dear Jim,
Thank you very much for your excellent explanations.
Basically I changed the file Burnouf.BhP.index.txt and added one more field (since we are not sure if we really use ipages, I put this field last).
[Bombay.BhP.index_copy.txt](https://github.com/user-attachments/files/18215706/Bombay.BhP.index_copy.txt)

I know Excel tabs quite well and now with your help I opened the CSV.

So what I did like a test:
1) test_plant_0.txt, test_plant_01.txt, test_plant_02.txt - these files have a CSV structure, and if I open them in Excel, I will see the data formatted in a table.
[test-plant_0.txt](https://github.com/user-attachments/files/18215710/test-plant_0.txt)
[test-plant_01.txt](https://github.com/user-attachments/files/18215712/test-plant_01.txt)
[test-plant_02.txt](https://github.com/user-attachments/files/18215713/test-plant_02.txt)

2) the file test_plants.xlsx [test_plants.xlsx](https://github.com/user-attachments/files/18215722/test_plants.xlsx)
 I saved as: 
a CSV table file [test_plants_1.csv](https://github.com/user-attachments/files/18215733/test_plants_1.csv)
a CSV text file [test_plants_2.txt](https://github.com/user-attachments/files/18215737/test_plants_2.txt)
a space-delimited text file [test_plants_3.txt](https://github.com/user-attachments/files/18215738/test_plants_3.txt)

Probably my version of Excel is quite old and it does not display the "tsv" format.

**Comment by funderburkjim** [comment]
@AnnaRybakovaT   I looked at test_plants_1.csv, test_planes_2.txt, and test_plants_3.txt
Any one of them looked fine.   I also copied test_plants.xlsx to google drive and opened there, and it also looks fine.

Also looked at Bombay.BhP.index_copy.txt, and compared with the pdf.  Everything looks
fine to me!

**Comment by AnnaRybakovaT** [comment]
> Also looked at Bombay.BhP.index_copy.txt, and compared with the pdf. Everything looks
> fine to me!

Thanks a lot!

**Comment by AnnaRybakovaT** [comment]
Dear all,
There are two files for the 1st and the 2nd volume. 
Please, look through some comments in the file Comments.docx

[Bombay.BhP.index.Vol1.txt](https://github.com/user-attachments/files/18321869/Bombay.BhP.index.Vol1.txt)
[Bombay.BhP.index.Vol2.txt](https://github.com/user-attachments/files/18321870/Bombay.BhP.index.Vol2.txt)

[Comments.docx](https://github.com/user-attachments/files/18321872/Comments.docx)

**Comment by AnnaRybakovaT** [comment]
Just for information,
 I will not available from tomorrow evening during 10 days. 
I will text when I arrive at home back.

**Comment by AnnaRybakovaT** [comment]
Dear all,
I'm back and ready for a new task. Until I get new information, I can continue with the BHS tooltips.

**Comment by Azanuka2412** [comment]
I'm ready to start indexing

**Comment by angalinde** [comment]
I'm ready to start indexing

**Comment by juliagabayraeva** [comment]
I'm ready to start indexing

**Comment by Andhrabharati** [comment]
@funderburkjim 

isn't this issue closable now?
